frontend/src/pages/articles/libs/helpers/get-article-tags.ts

```diff
@@ -0,0 +1,24 @@
+import { type ArticleWithAuthorType } from '~/packages/articles/articles.js';
+
+import { type TagType } from '../types/types.js';
+
+const getArticleTags = (article: ArticleWithAuthorType): TagType[] => {
+  const result: TagType[] = [];
+  if (article.genre) {
+    result.push({
+      id: crypto.randomUUID(),
+      category: 'genre',
+      text: article.genre,
+    });
+  }
+  if (article.prompt) {
+    for (const [key, value] of Object.entries(article.prompt)) {
+      if (value) {
+        result.push({ id: crypto.randomUUID(), category: key, text: value });
+      }
+    }
+  }
+  return result;
```

Comment:
And disable rule restricting `.reduce` usage
```suggestion
  const result: TagType[] = Boolean(article.genre)
     ? [{ category: 'genre', text: article.genre }]
     : [];
  
  return Object.entries(article.prompt ?? {}).reduce((resultingTags, [key, value]) => {
    if (value) {
      return resultingTags.concat({ category: key, text: value });
    }
    return resultingTags
  }, result); 
```

Reply:
type guard doesn't work with this variant 
![image](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/assets/44391531/aaade588-ec83-4dbe-be6e-59c8be30fa6e)

Reply:
Then just leave it as `article.genre` check